### ISSUE TITLE
Bot buying extra TP Tomes 

### ISSUE THREAD

**Issue by skyechotic** [kolton/d2bot-with-kolbot#1378]
My bot is setup using my own pickit file that is a combo of kolton.nip and a few other items ive added. It seems like throughout the run at some point he will buy an extra tome of tp and wont use the one that is already in his inventory. Do i need to add something to my pickit to make him stop? He also does this with keys.

**Comment by DarkHorseDre** [comment]
Pickit shouldn't affect this behaviour as by default the bot will detect a tomb and buy one if one isn't present.

just check that you haven't added a tome to pickit or haven't configured something else unnecessary.

if you're still stuck, post your town.js (if modified) and pickit to pastebin and share the link in a post here

**Comment by skyechotic** [comment]
My town.js hasnt been modified. I only touched my pally's profile and the dbjlead and my own pickit. The link for my pickit is https://pastebin.com/uEBsT1Sn

**Comment by DarkHorseDre** [comment]
Can you link your config?

**Comment by skyechotic** [comment]
Here is my characters config https://pastebin.com/ySR6Zdyf

**Comment by DarkHorseDre** [comment]
silly question - where in inv are you storing the tome?  in 0s or 1s space?

Config.Inventory[0] = [0,0,0,0,0,1,1,1,1,1];
    Config.Inventory[1] = [0,0,0,0,0,1,1,1,1,1];
    Config.Inventory[2] = [0,0,0,0,1,1,1,1,1,1];
    Config.Inventory[3] = [0,0,0,1,1,1,1,1,1,1];

**Comment by aim2kill** [comment]
@DarkHorseDre do3snt matter when you put your tomes at

**Comment by DarkHorseDre** [comment]
@aim2kill yeah I know, just working the problem as this shouldnt be happening..

@skyechotic aim2kill is right in that the only time the bot buys a tome is when it doesnt have one or it does cows.
tbh I would either download/version check all files (town.js and cows.js at least) or if they are correct I'd put a comment line in when it buys tomes to try and catch when it does it (assuming you have tried watching the bot do its runs? something you should do when troubleshooting for obvious reasons).

**Comment by skyechotic** [comment]
@DarkHorseDre I have watched it for several clears and he doesn't do it. I notice that when i go to bed and leave him running he will have on in his inventory when he running. However i did notice today after running through the night he didn't buy an extra one. Do you think its possible my game may have crashed/etc when he was going to make cows?

**Comment by aim2kill** [comment]
Probably

**Comment by DarkHorseDre** [comment]
Even if he lost one somehow, at the next town visit he would buy a tome. especially at the start of a run.

looking at town.js the bot will buy a tome if no tome and there is space.. assuming you are putting the tome where the 1's are it should be good. what is probably happening is the bot has no space to buy the new tome for cows and uses the existing one.. but I would save myself some time and download a separate new copy of the bot and create a new config (space + home) and use the same character - have it run the same bosses (manually set the bosses, don't reuse the config) and see if the problem reoccurs.

tbh the more i think of it, the more i think its likely the bot has no space for the tome in one of the runs - but let me know if you've disproven this.

from town.js:
`if (code === 518 && !me.findItem(518, 0, 3)) {
			tome = npc.getItem(518);

			if (tome && Storage.Inventory.CanFit(tome)) {
				try {
					tome.buy();
				} catch (e1) {
					print(e1);

					// Couldn't buy the tome, don't spam the scrolls
					return false;
				}`

if the new code download does the same i would watch the bot again and see how it deals with the tome.  i'd put some `D2Bot.printToConsole("comment");` lines in to capture any failures to buy tomes or situations where there are no tomes/cant open cows etc and narrow this down.
